kaitaistruct.py

```diff
@@ -320,9 +451,9 @@ def read_bytes_term(self, term, include_term, consume_term, eos_error):
     def ensure_fixed_contents(self, expected):
         actual = self._io.read(len(expected))
         if actual != expected:
-            raise Exception(
+            raise ValidationFailedError(
                 "unexpected fixed contents: got %r, was waiting for %r" %
-                (actual, expected)
+                (actual, expected), self._io, __file__
```

Comment:
They are still not machine readable. All the parameters should be put into exception object, not into the pregenerated str.